src/test/mir-opt/const_prop/cast.rs

```diff
@@ -0,0 +1,49 @@
+fn main() {
+    let x = 42u8 as u32;
```

Comment:
These already worked before this PR, right?

Reply:
Yes but we didn't have a test covering that.